### ISSUE TITLE
padding problem

### ISSUE THREAD

**Issue by yutaka329** [MPolaris/onnx2tflite#5]
Hi, nice work, 
 is there a method to remove pad before conv?
it may be fast without padding.

**Comment by MPolaris** [comment]
That's a good question.
Because tensorflow convolution padding is inconsistent with pytorch, tensorflow 'SAME' padding may pad more pixels on the right than left.
You can see https://stackoverflow.com/questions/52975843/comparing-conv2d-with-padding-between-tensorflow-and-pytorch.
If you get a good to fix this problem, welcome PR.

**Comment by yutaka329** [comment]
thanks, may be solved with some trick.

**Comment by yutaka329** [comment]
Hi
I  do some tests.
because network has many 3x3 conv with same padding.  
I change the code with below code  on mobileone.    it can reduce many pad and  get the same result,  reduce the peak of  mobile running time.  so we can use this trick^^.
      if pads is not None and max(pads) == 1 and max(strides) == 1:
            self.conv = keras.layers.DepthwiseConv2D(
                kernel_size, strides, "SAME", use_bias=False if bias is None else True,
                weights=[weights] if bias is None else [weights, bias],
                dilation_rate=dilations,
                activation=None,
                kernel_initializer='zeros',
                bias_initializer='zeros'
            )
            self.pad =None

**Comment by MPolaris** [comment]
> Hi I do some tests. because network has many 3x3 conv with same padding. I change the code with below code on mobileone. it can reduce many pad and get the same result, reduce the peak of mobile running time. so we can use this trick^^.
> 
> ```
>   if pads is not None and max(pads) == 1 and max(strides) == 1:
>         self.conv = keras.layers.DepthwiseConv2D(
>             kernel_size, strides, "SAME", use_bias=False if bias is None else True,
>             weights=[weights] if bias is None else [weights, bias],
>             dilation_rate=dilations,
>             activation=None,
>             kernel_initializer='zeros',
>             bias_initializer='zeros'
>         )
>         self.pad =None
> ```

Nice idea, thanks for your test.
Can you evaluate it on other types of convolution, such as keras.layers.Conv2D and keras.layers.Conv2DTranspose.
It's pleasure for your PR.

**Comment by yutaka329** [comment]
DepthwiseConv2D should be Consistent with Conv2D.    As to Conv2DTranspose, it need to be tested.

**Comment by MPolaris** [comment]
> 
Thanks for your nice work again, I will test it as soon~

**Comment by MPolaris** [comment]
I was tested it on this model, and your trick is work well, thanks for your contribution again.
```python
import torch
import torch.nn as nn

class testmodel(nn.Module):
    def __init__(self) -> None:
        super(testmodel, self).__init__()
        self.conv1 = nn.Conv2d(3, 6, 3, 2, 1)
        self.conv2 = nn.Conv2d(3, 6, 3, 1, 1)
        self.depthwiseConv1 = nn.Conv2d(3, 6, 3, 2, 1, groups=3)
        self.depthwiseConv2 = nn.Conv2d(3, 6, 3, 1, 1, groups=3)
        self.convTranspose1 = nn.ConvTranspose2d(6, 3, 2, 2, 0)
        self.convTranspose2 = nn.ConvTranspose2d(6, 30, 2, 2, 0)
        self.group_conv1 = nn.Conv2d(30, 30, 3, 1, 1, groups=30)

    def forward(self, X):
        x1 = self.conv1(X)
        x2 = self.conv2(X)

        x3 = self.depthwiseConv1(X)
        x4 = self.depthwiseConv2(X)
        
        x5 = self.convTranspose1(x1)
        x6 = self.convTranspose1(x3)

        out = x2+x4+torch.cat((x5, x6), dim=1)
        out = self.convTranspose2(out)
        out = self.group_conv1(out)
        return out

model = testmodel()
X = torch.randn(1, 3, 224, 224)
torch.onnx.export(model, X, './models/opt_test.onnx', opset_version=12)
```
